### PR TITLE
Add additional filters: `datetime_from_period` and `merge_date_url`

### PR DESCRIPTION
**Pull request by minchinweb**
A couple of tags I need to expand my [Seafoam theme](https://github.com/MinchinWeb/seafoam) to support period archive pages.

Also adds tests for all included filters. Also adds running tests to the GitHub workflow.

### CODE REVIEW

**Comment by minchinweb** [comment]
I'm not sure why the GitHub tests aren't running...

The code here is probably good to go, but I haven't actually done the theme development with them yet.... So no current rush to merge this while the other Pull Requests are still outstanding...

**Comment by justinmayer** [comment]
@MinchinWeb: Now that #8 has been merged, you'll need to rebase this PR onto `master`, resolving any conflicts in `tasks.py`.

**Comment by minchinweb** [comment]
@justinmayer We should be good to merge to master and cut a release.

I've rebased to master (and discovered/made use of some of what you'd mentioned earlier about [interactive rebasing](https://github.com/pelican-plugins/jinja-filters/pull/8#issuecomment-826326231)), fixed a few remaining bugs, and tested it locally, and everything looks good.

I got the GitHub Workflows working again although I had a fair bit of trouble (since resolved) with the linters as the version of black installed by poetry didn't match the version used by pre-commit. To help with that, I added a `--diif` pass-through option for `invoke lint` and enabled it for the Workflow.

**Comment by justinmayer** [comment]
@MinchinWeb: Do you want to update this PR with Black 21.4b0, which was just released 12 hours ago? 😅

**Comment by minchinweb** [comment]
@justinmayer *black* is updated! (It also requires bumping the minimum version of Python from 3.6.0 to 3.6.2).

**Comment by minchinweb** [comment]
(I think black is particularly prone to this because almost every update is an update to the "major" version number, which keeps poetry from automatically updating it.)

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,12 @@
 Jinja Filters Changelog
 =======================
 
+- **feature** add ``merge_date_url`` filter
+- **feature** add ``datetime_from_period`` filter
+- **support** rework supporting ``tasks.py`` (for use with ``invoke``)
+- **support** add test suite
+- **support** support Pelican from version 3.0 on
+
```
Comment (justinmayer) [inline CHANGELOG.rst]: The core design philosophy behind AutoPub includes never needing to manually add entries to the change-log. These entries should instead be in the form of a `RELEASE.md` file:
```markdown
Release type: patch

- **feature** add ``merge_date_url`` filter
- **feature** add ``datetime_from_period`` filter
- **support** rework supporting ``tasks.py`` (for use with ``invoke``)
- **support** add test suite
- **support** support Pelican from version 3.0 on
```

**Review by justinmayer — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst (hunk 1 above)
Comment (minchinweb) [reply CHANGELOG.rst]: @justinmayer Done! I've moved this to a `RELEASE.md` file.

**Review by justinmayer — APPROVED** [review]
Thanks for all the enhancements! ✨

**Pull request merged**